website/source/guides/tls.html.md

```diff
@@ -0,0 +1,425 @@
+without presenting a client-side certificate. If `verify_https_client` is
+enabled ony HTTP API clients presenting a certificate signed by the same CA as
+Nomad's certificate are allowed to access Nomad.
+
+~> Enabling `verify_https_client` feature effectively protects Nomad from
+   unauthorized network access at the cost of breaking compatibility with Consul
+   HTTPS health checks.
+
+### Client Configuration
+
+The Nomad client configuration is similar to the server configuration. The
+biggest difference is in the certificate and key used for configuration.
+
+```hcl
+# Increase log verbosity
+log_level = "DEBUG"
+
+# Setup data dir
+data_dir = "/tmp/client1"
+
+# Enable the client
+client {
+  enabled = true
+
+  # For demo assume we are talking to server1. For production,
+  # this should be like "nomad.service.consul:4647" and a system
+  # like Consul used for service discovery.
+  servers = ["127.0.0.1:4647"]
+}
+
+# Modify our port to avoid a collision with server1
+ports {
+  http = 5656
+}
+
+# Require TLS
+tls {
+  http = true
+  rpc  = true
+
+  ca_file   = "nomad-ca.pem"
+  cert_file = "client.pem"
+  key_file  = "client-key.pem"
+
+  verify_server_hostname = true
+  verify_https_client    = true
+}
+```
+
+### Running with TLS
+
+Now that we have certificates generated and configuration for a client and
+server we can test our TLS-enabled cluster!
+
+In separate terminals start a server and client agent:
+
+```shell
+$ # In one terminal...
+$ nomad agent -config server1.hcl
+
+$ # ...and in another
+$ nomad agent -config client1.hcl
+```
+
+If you run `nomad node-status` now, you'll get an error, like:
+
+```text
+Error querying node status: Get http://127.0.0.1:4646/v1/nodes: malformed HTTP response "\x15\x03\x01\x00\x02\x02"
+```
+
+This is because the Nomad CLI defaults to communicating via HTTP instead of
+HTTPS. We can configure the local Nomad client to connect using TLS and specify
+our custom keys and certificates using the command line:
+
+```shell
+$ nomad node-status -ca-cert=nomad-ca.pem -client-cert=cli.pem -client-key=cli-key.pem -addr=https://127.0.0.1:4646
+```
+
+This process can be cumbersome to type each time, so the Nomad CLI also
+searches environment variables for default values. Set the following
+environment variables in your shell:
+
+```shell
+$ export NOMAD_ADDR=https://localhost:4646
+$ export NOMAD_CACERT=nomad-ca.pem
+$ export NOMAD_CLIENT_CERT=cli.pem
+$ export NOMAD_CLIENT_KEY=cli-key.pem
+```
+
+* `NOMAD_ADDR` is the URL of the Nomad agent and sets the default for `-addr`.
+* `NOMAD_CACERT` is the location of your CA certificate and sets the default
+  for `-ca-cert`.
+* `NOMAD_CLIENT_CERT` is the location of your CLI certificate and sets the
+  default for `-client-cert`.
+* `NOMAD_CLIENT_KEY` is the location of your CLI key and sets the default for
+  `-client-key`.
+
+After these environment variables are correctly configured, the CLI will
+respond as expected:
+
+```text
+$ nomad node-status
```

Comment:
Looks like that's probably the same bug as #2490. I filed #3013 and will try to get it fixed for 0.6.1.

Thanks for testing it out!